modules/core/src/v2/coins/dot.ts

```diff
@@ -221,6 +210,27 @@ export class Dot extends BaseCoin {
     return { txHex: signedTxHex };
   }
 
+  /**
+   * Explain/parse transaction
+   * @param params
+   * @param callback
+  */
+  async explainTransaction(params: ExplainTransactionOptions): Promise<accountLib.Dot.TransactionExplanation> {
+    const txHex = params.txPrebuild.txHex;
+    if (!txHex || !params.txPrebuild.validity || !params.txPrebuild.key || !params.txPrebuild.referenceBlock || !params.txPrebuild.version) {
+      throw new Error('missing explain tx parameters');
+    }
+    const sender = new accountLib.Dot.KeyPair({ pub: params.txPrebuild.key }).getAddress();
+    const factory = accountLib.register(this.getChain(), accountLib.Dot.TransactionBuilderFactory);
+    const txBuilder = factory.from(txHex);
+    txBuilder
+      .validity(params.txPrebuild.validity)
+      .referenceBlock(params.txPrebuild.referenceBlock)
+      .sender({ address: sender })
+      .version(params.txPrebuild.version);
```

Comment:
Aren't these in the txHex already?

Reply:
No they have to be set beforehand. They cannot be decoded from the hex.